src/tool/hpcrun/sample-sources/perf/kernel_blocking.c

```diff
@@ -126,9 +128,11 @@ blame_kernel_time(event_thread_t *current_event, cct_node_t *cct_kernel,
   // ----------------------------------------------------------------
   // it's important to always count the number of samples for debugging purpose
   // ----------------------------------------------------------------
-
-  thread_data_t *td = hpcrun_get_thread_data();
-  td->core_profile_trace_data.perf_event_info[metric_blocking_index].num_samples++;
+// fix issue #556: since we use blocktime metric kind, we don't have access to perf info
```

Comment:
So perf's metrics kind cannot be used by blocktime? Sorry I don't entirely understand the comment.

Reply:
@jmellorcrummey The first commit is wrong. I thought it was the solution.
@Jokeren The number of samples is not the same as the metric value.  Sometimes I need to know the number of samples to check the correctness of the metric value. 
The blocktime event is a "derived" event of perf event which uses perf's context switch event. It shouldn't have access to any perf information.